google/cloud/bigquery/dataset.py

```diff
@@ -937,6 +940,24 @@ def _build_resource(self, filter_fields):
         """Generate a resource for ``update``."""
         return _helpers._build_resource_from_properties(self, filter_fields)
 
+    @property
+    def external_catalog_dataset_options(self):
+        """Options defining open source compatible datasets living in the
+        BigQuery catalog. Contains metadata of open source database, schema
+        or namespace represented by the current dataset."""
+
+        return self._properties.get("externalCatalogDatasetOptions")
+
+    @external_catalog_dataset_options.setter
+    def external_catalog_dataset_options(self, value):
+        if not isinstance(value, ExternalCatalogDatasetOptions) and value is not None:
+            raise ValueError(
+                "external_catalog_dataset_options must be an "
+                "ExternalCatalogDatasetOptions object or None. "
+                f"Got {repr(value)}."
```

Comment:
NIT

(a really small one) Use `type(value)` to make the error message more explicit. 
```suggestion
                f"Got {type(value)}."
```

Reply:
Cool. I will make these changes from `repr` to `type`.

Reply:
It looks like this didn't get updated.

This looks like another case where the _isinstance_or_raise helper can be used as well.

I'm also a little confused with why this case raises a ValueError and the helper raises a TypeError, is there something different about the case here?

Reply:
To answer your question: 

This is a moving target. Thus after this bit of code was made, we eventually migrated to using a function to handle the same/similar capability. But before going back and changing all the things, I felt I needed to get the `isinstance_or_raise()` function nailed down. Somewhere in that process we looked at what the appropriate error ought to be and I assert that it should be a TypeError (but mistakenly thought is was a ValueError when I started out).

I think `isinstance_or_raise()` is mostly nailed down now and can be used in future classes and can also replace older methods I created way back when this task first started.

sigh.

= /